test/test_CompactSyntaxParser.py

```diff
@@ -125,7 +125,7 @@ def test_parsing_simple_expressions(self):
                           [[['1', '<', '2'], ['3', '+', '4'], ['5', '*', '6']]])
 
     # XML-Only method?
-    # def testParsingTrace(self):
+    # def test_parsing_trace(self):
```

Comment:
Yeah, this is XML-only and can be removed. The Python implementation can trace naturally using the stack, so doesn't need special support!

Reply:
Do we still need the `class Trace` and other tracing methods then?

Reply:
Ah, sorry, I misremembered. There is still something that happens in the Python implementation with this - it sets `wrapped_expr.trace = True`, which triggers a call to `Locatable.trace_value` when the expression is evaluated. It's a debugging aid, though probably isn't used much!

You could enhance the test with a check like this?
```python
action = grammar.parseString('var?', parseAll=True)
assert action.expr().trace

action = grammar.parseString('var', parseAll=True)
assert not action.expr().trace
```

Reply:
So what would I do with the PROTO_NS bit?

```
    def testParsingTrace(self):
       trace = {'{%s}trace' % CSP.PROTO_NS: '1'}
       self.assertParses(csp.expr, '1?', [['1']])
       self.assertParses(csp.expr, 'var?', [['var']])
       self.assertParses(csp.expr, '(1 + a)?', [[['1', '+', 'a']]])
       self.assertParses(csp.expr, '1 + a?', [['1', '+', ['a']]])
```

Reply:
That bit can go - it was just checking that the XML conversion added the attribute.

Reply:
```
        action = csp.expr.parseString('var?', parseAll=True)
        print(action)
        print(type(action))
        assert action._expr().trace
```
I've tried this but that returns:
```
[Trace[Variable[var]]]
<class 'pyparsing.ParseResults'>
```
and then
```
    assert action._expr().trace
E   TypeError: 'str' object is not callable
```

Reply:
Maybe you need `action[0].expr().trace` then? Looks like the `Action` class is wrapped from the outer `[]`